smk/QC_0.smk

```diff
@@ -31,72 +33,65 @@ use rule QC_0_base, QC_0_rpkg from print_versions as version_*
 
 snakefile_name = print_versions.get_smk_filename()
 
-if config['raw_reads_dir'] is None:
-    print('ERROR in ', config_path, ': raw_reads_dir variable in configuration yaml file is empty. Please, complete ', config_path)
-else:
-    raw_dir = config['raw_reads_dir']
-
-if config['FASTP_threads'] is None:
-    print('ERROR in ', config_path, ': FASTP_threads variable is empty. Please, complete ', config_path)
-elif type(config['FASTP_threads']) != int:
-    print('ERROR in ', config_path, ': FASTP_threads variable is not an integer. Please, complete ', config_path)
+raw_dir              = validate_required_key(config, 'raw_reads_dir')
+perc_remaining_reads = validate_required_key(config, 'perc_remaining_reads')
 
-if config['FASTP_memory'] is None:
-    print('ERROR in ', config_path, ': FASTP_memory variable is empty. Please, complete ', config_path)
-elif type(config['FASTP_memory']) != int:
-    print('ERROR in ', config_path, ': FASTP_memory variable is not an integer. Please, complete ', config_path)
+FASTP_threads        = validate_required_key(config, 'FASTP_threads')
+FASTP_memory         = validate_required_key(config, 'FASTP_memory')
+FASTP_min_length     = validate_required_key(config, 'FASTP_min_length')
+FASTP_front_mean_qual= validate_required_key(config, 'FASTP_front_mean_qual')
+FASTP_tail_mean_qual = validate_required_key(config, 'FASTP_tail_mean_qual')
 
-if config['FASTP_adapters'] in ('Skip', 'Quality', 'Overlap', 'Detect'):
+FASTP_adapters       = validate_required_key(config, 'FASTP_adapters')
```

Comment:
Could `FASTP_adapters` type be `['file', str]` and location checked via `config_parser`?

Reply:
This could be done, yes. But we will still need to do a conditional on `os.path.exists` and if they are strings then ensure they are one of 4 allowed values. So this check for file cannot be avoided anyway.  It will look like:
```
FASTP_adapters       = validate_required_key(config, 'FASTP_adapters')
if not os.path.exists(FASTP_adapters):
    check_allowed_values('FASTP_adapters', FASTP_adapters, ('Skip', 'Quality', 'Overlap', 'Detect'))
```

May be this is more elegant? If yes, I can implement this for both adapters.